peerconnection.go

```diff
@@ -1787,14 +1795,20 @@ func (pc *PeerConnection) generateUnmatchedSDP(useIdentity bool) (*sdp.SessionDe
 		mediaSections = append(mediaSections, mediaSection{id: strconv.Itoa(len(mediaSections)), data: true})
 	}
 
-	return populateSDP(d, isPlanB, pc.api.settingEngine.candidates.ICELite, pc.api.mediaEngine, connectionRoleFromDtlsRole(defaultDtlsRoleOffer), candidates, iceParams, mediaSections, pc.ICEGatheringState())
+	var mediaCert *Certificate;
+
+	if (pc.configuration.SDPMediaLevelFingerprints) {
+		mediaCert = &pc.configuration.Certificates[0]
+	}
+	
+	return populateSDP(d, isPlanB, mediaCert, pc.api.settingEngine.candidates.ICELite, pc.api.mediaEngine, connectionRoleFromDtlsRole(defaultDtlsRoleOffer), candidates, iceParams, mediaSections, pc.ICEGatheringState())
 }
 
 // generateMatchedSDP generates a SDP and takes the remote state into account
 // this is used everytime we have a RemoteDescription
 func (pc *PeerConnection) generateMatchedSDP(useIdentity bool, includeUnmatched bool, connectionRole sdp.ConnectionRole) (*sdp.SessionDescription, error) {
 	d := sdp.NewJSEPSessionDescription(useIdentity)
-	if err := addFingerprints(d, pc.configuration.Certificates[0]); err != nil {
+	if err := addSessionFingerprints(d, pc.configuration.Certificates[0]); err != nil {
```

Comment:
What do you think of keeping `addFingerprints` but then have a third argument `SDPMediaLevelFingerprints`. Will make testing a lot easier I think.

I am just worried that having multiple places check `SDPMediaLevelFingerprints` will cause people to mess things up.

Reply:
I agree, centralizing is probably a good idea, i'll take another pass

Reply:
I've actually factored out the `addFingerprints` function entirely, the decision gets made in `populateSDP`. It's not safe to call `addFingerprints` more then once anyway, so this is a bit safer. I've messed something up in my refactor, and some tests are failing. I'll push a new commit soon once i figure that out. I think this is a definite improvement.